### DISCUSSION TITLE
Valetudo 2023.08.0

### DISCUSSION THREAD

**Discussion by Hypfer** [Hypfer/Valetudo#1855] [Release Discussions]
<div align="center">
    <img src="https://raw.githubusercontent.com/Hypfer/Valetudo/master/assets/logo/valetudo_logo_with_name.svg" width="800" alt="valetudo">
    <p align="center"><h2>2023.08.0</h2></p>
</div>

DEFCON 31 - 7 newly rootable robots, new rooting method for 6 robots with a previous root, 5 more new robots in the pipeline

## DEFCON 31

If you haven't seen it already, I highly recommend watching our latest talk:<br/>
https://media.ccc.de/v/camp2023-57158-vacuum_robot_security_and_privacy (CCCamp 2023 version)

The Valetudo docs have already been partially updates with new rooting methods and robot details. It will take some more time to fully finish all documentation. Please bear with me :)

The new Dreame Rooting PCB can be found here: https://github.com/Hypfer/valetudo-dreameadapter

As a spoiler, here's the full list of what we just released in that talk:

### Newly rootable robots

These robots are now rootable for the first time

- Dreame L10s Ultra
- Roborock S8
- Roborock S8 Pro Ultra
- Xiaomi X10+
- Dreame D10s Plus
- Dreame D10s Pro

### Robots with a new rooting method

These robots have been rootable previously but received firmware patches that blocked the old methods

- Dreame Z10
- Dreame L10
- Dreame W10
- Dreame D9
- Dreame F9
- Xiaomi Vacuum-Mop 2 Ultra

### Robots currently in the pipeline

**Edit from the future (2024-04):**
> Since this text keeps haunting me, I need to add this note.
> 
> It didn't work out.
> 
> I've got way too hyped and wanted to help too many people and with that ended up spending more than 4000€ of my own hard-earned cash on dead-ends that led nowhere. Please stop asking about these or robots mentioned in the section above, okay?
> It is frustrating enough as it is. No I did not accidentally forget to update the docs. No it's not just a tiny step away.
> 
> I accidentally tried to be nice, wasting a lot of my own personal resources on that, and now the reminder of that waste of time and resources periodically pops up in my inbox as a "but but but you said that".
> And yes, this also includes robots that might've been mentioned in other release notes. Let it go.

We're currently finishing up the rooting methods for these robots. It might take a little while as it's conference season

- Dreame L10 Ultra
- Dreame L10s Pro
- Dreame W10 Pro
- Roborock Q Revo
- Roborock S7 Max Ultra

## Valetudo features

While most of our time was spent getting the fresh rooting methods ready, a few things extending beyond adding support
for the new robots also happened to Valetudo

### Return of the ObstacleAvoidanceCapability

With the Roborock S8 now being officially rootable and supported, toggling the obstacle avoidance on or off was reintroduced 
as a full capability again after it was previously demoted to being just a dreame-specific quirk.

![image](https://github.com/Hypfer/Valetudo/assets/974410/3cd8a5c8-1ba0-4850-b359-5311403b7a53)

Furthermore, the `PetObstacleAvoidanceControlCapability` was introduced as a way to tell the Obstacle Detection AI to be especially
careful with some particularly dangerous kinds of obstacles.

### Even more capabilities

With the `CarpetSensorModeControlCapability` there's now proper UI to select how a robot equipped with a dedicated carpet
sensor should deal with detected carpet. This feature was previously found in the quirks section.

![image](https://github.com/Hypfer/Valetudo/assets/974410/59ab2c7e-4c6a-4d3b-bb50-9cb3ad13a89d)

The last new capability for this release is the `CollisionAvoidantNavigationControlCapability` which toggles what the name suggests. 

![image](https://github.com/Hypfer/Valetudo/assets/974410/7651ccad-9e27-4b79-ba8b-07fe6df22ead)

#### New Entities

For convenience and better automations, Battery Level and Status Flag are now being exposed to Home Assistant as separate
entities. Previously, the Battery Level was only available as an attribute of the Vacuum entity.

![image](https://github.com/Hypfer/Valetudo/assets/974410/f6b92a1f-d359-4543-a392-53b8ce093c38)

#### Note regarding Home Assistant 2023.08 and newer

Valetudo users that have upgraded their Home Assistant Instance to 2023.8 are currently being confronted with a scary error message that looks like this:

![image](https://github.com/Hypfer/Valetudo/assets/974410/1c76e8eb-d4d0-47a3-a585-036d36b808e2)

This is neither an actual functional issue nor a bug in Valetudo.

There were some architectural and strategic changes inside Home Assistant that affect some details about how MQTT Autodiscovery
should be implemented by other software that wants to show up in Home Assistant. Implementing the requested changes not very hard,
however doing so would break the naming of Valetudo-related entities in older versions of Home Assistant.

Since the release of HA 2023.8 happened less than two weeks ago, it would be unreasonable to force all users of Valetudo
to upgrade their HA instances. Especially since due to migration logic in HA, things work just fine and will continue to work like that until 02/2024.

For now, simply click on the "Ignore" Button in Home Assistant. No need to rename anything.

## The usual

I am quite happy with what we've released today and I think you will be too :)

If you want to see Valetudo on more robots and/or like this release, you might want to consider donating:

[https://github.com/sponsors/Hypfer](https://github.com/sponsors/Hypfer)

[https://builder.dontvacuum.me/donations.txt](https://builder.dontvacuum.me/donations.txt)

### Features

- **ui**: Include consumable name in consumable reset confirm dialog [`7f7b218`](http://github.com/Hypfer/Valetudo/commit/7f7b218001c4355ae069e31bb9f30786f0f5daf4)
- **mqtt**: Expose battery state and status flag as home assistant sensor entities [`5f26145`](http://github.com/Hypfer/Valetudo/commit/5f2614580ed7390ea8467174c310a36b8ab99709)
- **vendor.roborock**: S8 Pro Ultra support [`0f390e0`](http://github.com/Hypfer/Valetudo/commit/0f390e0c6dfdee6cd1773765881906e2d355fbf1)
- **vendor.viomi**: Support outline cleaning mode (#1844) [`f68b995`](http://github.com/Hypfer/Valetudo/commit/f68b99586f69a7644f75cf4fff1e33ccd3f4a5ef)
- CarpetSensorModeControlCapability [`a4261b6`](http://github.com/Hypfer/Valetudo/commit/a4261b641e21a3fa8674c68042f1114a21181065)
- CollisionAvoidantNavigationControlCapability [`e22e532`](http://github.com/Hypfer/Valetudo/commit/e22e532982673fcc95ec6ece9740e078626ac51e)
- **vendor.dreame**: More quirks [`5a3ecc6`](http://github.com/Hypfer/Valetudo/commit/5a3ecc64c5f5499bb9cf2b43ce1e22850fb4622c)
- PetObstacleAvoidanceControlCapability [`7eae549`](http://github.com/Hypfer/Valetudo/commit/7eae549ee55331ddf888383cd3d0b83ab7421fb2)
- **vendor.roborock**: RoborockObstacleAvoidanceControlCapability [`bd1febe`](http://github.com/Hypfer/Valetudo/commit/bd1febe337192a31185a1bb316793d029da4abff)
- Return of the ObstacleAvoidanceControlCapability [`e48aa7e`](http://github.com/Hypfer/Valetudo/commit/e48aa7e78de3ddd8d27056fe15dc61387f15b89b)
- **vendor.roborock**: S8 support + map parser cleanup [`12aa3a7`](http://github.com/Hypfer/Valetudo/commit/12aa3a73371cfd55d6ffa0a99575ba19ae523e73)
- **vendor.dreame**: L10 Ultra [`4e4b7ae`](http://github.com/Hypfer/Valetudo/commit/4e4b7ae2641fff08bb91197b704e4ebf977901c9)

### Fixes

- **vendor.roborock**: Remove invalid quirk for S8 series [`76bba87`](http://github.com/Hypfer/Valetudo/commit/76bba87a3976bdfefec899eb7d8d9b84449e7bba)
- **vendor.dreame**: The L10S Pro does not support the CollisionAvoidantNavigationControlCapability [`c3405bd`](http://github.com/Hypfer/Valetudo/commit/c3405bdb8ebbd930a21835152211b2e8363da705)

### Refactoring

- **vendor.dreame**: Hard-code miot IDs for capabilities that are exclusive to the gen2 miot schema [`ef93adf`](http://github.com/Hypfer/Valetudo/commit/ef93adf2d5259e7537f1983eed2848e0df030ef8)

### Chores

- **release**: 2023.08.0 [`ed051c6`](http://github.com/Hypfer/Valetudo/commit/ed051c69a5decfc8db09e3c829a5077ad68b290a)
- **vendor.dreame**: Minor cleanup of confusing loglines [`eb73525`](http://github.com/Hypfer/Valetudo/commit/eb7352565f7c55d2dd783fedb1d2f9c7a2abc166)
- **build**: Include openAPI spec json as a release asset for prod and nightly [`db9f42e`](http://github.com/Hypfer/Valetudo/commit/db9f42ea8c0e727108a9680fc3c825753fd20131)

<hr /><em>This discussion was created from the release <a href='https://github.com/Hypfer/Valetudo/releases/tag/2023.08.0'>Valetudo 2023.08.0</a>.</em>

**Comment by elec2** [comment]
Hey, I just discovered the project and I'm so hyped to finally be able to cut my Q Revo from the net. Really looking forward for the support, and maybe even be able to use it as a mobile camera?!  Omg this is so great! Thank you very much!

**Reply by Hypfer** [reply]
Sorry, I was too optimistic back when I wrote these release notes in august of 2023.
The Q Revo proved harder to root than anticipated. It probably won't ever happen.

Not sure where you came up with that "mobile camera" stuff but that's also not going to happen.
